engine/security/src/main/scala/pl/touk/nussknacker/ui/security/oidc/OidcAuthenticationProvider.scala

```diff
@@ -0,0 +1,25 @@
+package pl.touk.nussknacker.ui.security.oidc
+
+import com.typesafe.config.Config
+import pl.touk.nussknacker.ui.security.api.AuthenticationProvider
+import pl.touk.nussknacker.ui.security.oauth2.OAuth2AuthenticationResources
+import sttp.client.{NothingT, SttpBackend}
+
+import scala.concurrent.{ExecutionContext, Future}
+
+
+class OidcAuthenticationProvider extends AuthenticationProvider {
+
+  import net.ceedubs.ficus.readers.ArbitraryTypeReader._
+  import pl.touk.nussknacker.engine.util.config.CustomFicusInstances._
+  import pl.touk.nussknacker.ui.security.api.AuthenticationConfiguration.authenticationConfigPath
+
+  def name: String = "Oidc"
+
+  override def createAuthenticationResources(config: Config, classLoader: ClassLoader)(implicit ec: ExecutionContext, sttpBackend: SttpBackend[Future, Nothing, NothingT]): OAuth2AuthenticationResources = {
+    new OidcAuthenticationResources(
+      realm,
+      config.as[OidcAuthenticationConfiguration](authenticationConfigPath).withDiscovery
```

Comment:
do I understand correctly that right now we'll try to discover config based on issuer even though authorizationEndpoint is provided ? do we want it to work that way ?

Reply:
Yes.
OpenID Discovery provides configuration for any endpoint required in the flow. We cannot skip querying if only one endpoint is provided in the local configuration to override the remote one. And check if all of them are provided seems to be a very improbable case.